generate-routes.ts

```diff
@@ -0,0 +1,453 @@
+import { readFile, writeFile } from 'node:fs/promises'
+import { resolve } from 'node:path'
+
+import { openapi } from '@seamapi/types/connect'
+import { camelCase, paramCase, pascalCase, snakeCase } from 'change-case'
+import { ESLint } from 'eslint'
+import { format, resolveConfig } from 'prettier'
+
+const rootClassPath = resolve('src', 'lib', 'seam', 'connect', 'client.ts')
+const routeOutputPath = resolve('src', 'lib', 'seam', 'connect', 'routes')
+
+const routePaths = [
+  '/access_codes',
+  '/access_codes/unmanaged',
+  '/acs',
+  '/acs/access_groups',
+  '/acs/credentials',
+  '/acs/systems',
+  '/acs/users',
+  '/action_attempts',
+  '/client_sessions',
+  '/connect_webviews',
+  '/connected_accounts',
+  '/devices',
+  '/devices/unmanaged',
+  '/events',
+  '/locks',
+  '/noise_sensors',
+  '/noise_sensors/noise_thresholds',
+  '/thermostats/climate_setting_schedules',
+  '/thermostats',
+  '/webhooks',
+  '/workspaces',
+] as const
+
+const routePathSubresources: Partial<
+  Record<(typeof routePaths)[number], string[]>
+> = {
+  '/access_codes': ['unmanaged'],
+  '/acs': ['access_groups', 'credentials', 'systems', 'users'],
+  '/devices': ['unmanaged'],
+  '/noise_sensors': ['noise_thresholds'],
+  '/thermostats': ['climate_setting_schedules'],
+}
+
+const ignoredEndpointPaths = [
+  '/access_codes/simulate/create_unmanaged_access_code',
+  '/health',
+  '/health/get_health',
+  '/health/get_service_health',
+  '/health/service/[service_name]',
+  '/noise_sensors/simulate/trigger_noise_threshold',
+] as const
+
+const endpointResources: Partial<
+  Record<
+    keyof typeof openapi.paths,
+    | 'action_attempt'
+    | 'access_codes'
+    | 'device_providers'
+    | 'generated_code'
+    | 'backup_access_code'
+    | 'acs_users'
+    | null
+  >
+> = {
+  '/access_codes/delete': null,
+  '/access_codes/unmanaged/delete': null,
+  '/access_codes/update': null,
+  '/connect_webviews/view': null,
+  '/noise_sensors/noise_thresholds/create': null,
+  '/noise_sensors/noise_thresholds/delete': null,
+  '/noise_sensors/noise_thresholds/update': null,
+  '/thermostats/climate_setting_schedules/update': null,
+  '/workspaces/reset_sandbox': null,
+} as const
```

Comment:
These override the default return value for an endpoint when `deriveResource` would otherwise detect the wrong thing.